framework/meta-lib/src/contract/sc_config/contract_variant_builder.rs

```diff
@@ -253,40 +254,112 @@ fn validate_contract_variants(contracts: &[ContractVariant]) {
     }
 }
 
-impl ScConfig {
-    /// Assembles an `ContractVariantConfig` from a raw config object that was loaded via Serde.
-    ///
-    /// In most cases the config will be loaded from a .toml file, use `load_from_file` for that.
-    pub fn load_from_config(config: &ScConfigSerde, original_abi: &ContractAbi) -> Self {
-        let mut contract_builders: HashMap<String, ContractVariantBuilder> = config
-            .contracts
-            .iter()
-            .map(ContractVariantBuilder::map_from_config)
-            .collect();
+fn process_contracts(config: &ScConfigSerde, original_abi: &ContractAbi) -> Vec<ContractVariant> {
+    let mut contract_builders: HashMap<String, ContractVariantBuilder> = config
+        .contracts
+        .iter()
+        .map(ContractVariantBuilder::map_from_config)
+        .collect();
+
+    collect_and_process_endpoints(
+        &mut contract_builders,
+        original_abi,
+        &config.labels_for_contracts,
+    );
+
+    let mut contracts: Vec<ContractVariant> = contract_builders
+        .into_values()
+        .map(|builder| build_contract(builder, original_abi))
+        .collect();
+
+    if contracts.is_empty() {
+        contracts.push(ContractVariant::default_from_abi(original_abi));
+    }
+    set_main_contract_flag(&mut contracts, &config.settings.main);
+    validate_contract_variants(&contracts);
+
+    contracts
+}
+
+fn process_proxy_contracts(
+    config: &ScConfigSerde,
+    original_abi: &ContractAbi,
+) -> HashMap<ProxyConfigSerde, ContractVariant> {
```

Comment:
Should be a Vec

Reply:
The config should not be the key here, it should be part of the proxy config
You need to add the ContractVariant to the ProxyConfig